<?xml version="1.0" encoding="UTF-8" standalone="yes" ?>
<!DOCTYPE bugzilla SYSTEM "https://bugs.eclipse.org/bugs/page.cgi?id=bugzilla.dtd">

<bugzilla version="4.4.1"
          urlbase="https://bugs.eclipse.org/bugs/"
          
          maintainer="webmaster@eclipse.org"
>

    <bug>
          <bug_id>214257</bug_id>
          
          <creation_ts>2008-01-03 12:59:00 -0500</creation_ts>
          <short_desc>Variable declaration in catch is declared duplicate but should have new scope</short_desc>
          <delta_ts>2010-03-03 11:16:31 -0500</delta_ts>
          <reporter_accessible>1</reporter_accessible>
          <cclist_accessible>1</cclist_accessible>
          <classification_id>8</classification_id>
          <classification>WebTools</classification>
          <product>JSDT</product>
          <component>General</component>
          <version>unspecified</version>
          <rep_platform>PC</rep_platform>
          <op_sys>Windows XP</op_sys>
          <bug_status>RESOLVED</bug_status>
          <resolution>FIXED</resolution>
          
          
          <bug_file_loc></bug_file_loc>
          <status_whiteboard></status_whiteboard>
          <keywords></keywords>
          <priority>P3</priority>
          <bug_severity>normal</bug_severity>
          <target_milestone>3.0 M5</target_milestone>
          
          
          <everconfirmed>1</everconfirmed>
          <reporter name="David Thompson">dthompson</reporter>
          <assigned_to name="Phil Berkland">berkland</assigned_to>
          <cc>dthompson</cc>
          <qa_contact name="Phil Berkland">berkland</qa_contact>
          <votes>0</votes>

      

      

      

          <comment_sort_order>oldest_to_newest</comment_sort_order>  
          <long_desc isprivate="0" >
    <commentid>1125107</commentid>
    <comment_count>0</comment_count>
    <who name="David Thompson">dthompson</who>
    <bug_when>2008-01-03 12:59:50 -0500</bug_when>
    <thetext>The following code demonstrates the problem. The second e is flagged as 
Duplicate parameter e, but I believe this should be valid and the e within the catch should have a different scope than the declared e.

var e = 5;
function main3() {
    try {
       var doSomething = 1;   
    } catch (e) {
        doSomething = 2;
    }
}</thetext>
  </long_desc><long_desc isprivate="0" >
    <commentid>1128308</commentid>
    <comment_count>1</comment_count>
    <who name="Phil Berkland">berkland</who>
    <bug_when>2008-01-09 13:10:06 -0500</bug_when>
    <thetext>Fixed so will not show duplicate error, but will still show configurable warning that other var is hidden.  </thetext>
  </long_desc>
      
      

    </bug>

</bugzilla>